protokolle/29-04-2020_protokoll.md

```diff
@@ -0,0 +1,262 @@
+
+    *  semistrukturierte Daten im JSON-Format (Folie 13) --> *Prüfungsfragen: Welche Datenstruktur ist das? Welches Datenformat haben wir?*
+
+    *  **COMMENT ON TABLE** SCHEMA.TABELLE - hängt die Metadaten an eine Tabelle
+
+3. Verknüpfen von Tabellen (JOIN)
+
+    * Fragestellungen: Welche Informationen sind in einer Tabelle enthalten und in der anderen nicht?
+
+    * Verknüpfungsmuster: INNER-, LEFT-, FULL-JOIN sind die wichtigsten und RIGHT, NATURAL und CROSS gibt es auch
+
+    * Aufgabe: die richtigen Datenbankbefehle für die Fehler -> Lücken finden und Lücken schließen
+
+    * Programmier-Code siehe Folie 16
+
+    * Beispiel für einen LEFT-JOIN: Zeitreihe mit Wetter verknüpfen
+
+4. Arbeitsschritte
+
+    * siehe Folie 21 und weitere
+
+    * SELECT (welche Spalten) FROM (SCHEMA.TABELLE)  WHERE (Bedingung)
+ 
+
+
+# 16:30 Datenbanken II
+
+## pg-Admin4:
+
+	* Web-GUI, die SQL-code ausführt
+	
+	* Man verbindet sich immer mit einem Datenbank-Server und nicht direkt mit einer Datenbank
+    
+
+    * Servergruppe erstellen: Servers -> create -> Server Group bspw. HTW-PV3
+
+    * Server erstellen: HTW-PV3 -> create -> Server
+
+      * Reiter General: Name = PostgresSQL 12
+
+      * Reiter Connection: Host name/address=localhost
+
+      * save
+
+    * Server wählen: PostgresSQL 12
+
+      * postgres-Datenbank nicht anfassen außer die ersten beiden Skripte?!
+
+
+    * Ein Befehl der ausgeführt wird, wird immer nur auf einer Datenbank ausgeführt
+
+      * Fehlerquelle: Falsche DB ausgewählt und code somit auf falscher DB ausgeführt (Vorsicht!)
+
+    * *Wie führe ich auch einer Datenbank code aus?*
+
+      1. * Server auswählen
+
+      2. * Datenbank auswählen 
+
+      3. * Rechtskick: Querry-tool:
+
+         * Code einfügen und ausführen (F5)
+
+         * Querry wird immer mit Ausgabe "Query returned successfully in XXX msec." abgeschlossen
+
+         * Falls nicht, wurde nichts ausgeführt (All or nothing)
+
+         * **!Wichtig!** In der Regel muss ausgeführter CODE (rechts) im Browser (links) per Rechtsklick>Refresh neugeladen werden.
+
+      4. * Disconnected vom Server
+	      
+      5. * PostgresSQL 12: Properties ändern
+
+         * user == sonnja
+
+      6. * reconnecten mit Passwort : sonnja
+
+
+    * *FRAGE: Wie wurde die Datenbank "sonnja_db" hinzugefügt? Steht das in dem Skript 2?*
+
+    * -> Ja (Aber Prinzipiell Rechtsklick auf die Datenbank und neue Datenbank erzeugen)
+
+
+
+	* User Management mit Skript (von LH) erstellen:
+
+      * user Rolle erstellen
+
+      * admin Rolle erstellen
+
+
+# 17:00 Daten-Session
+
+    Welche Daten haben wir?
+
+    METADATEN erstellen
+
+    Datenbankimport versuchen
+
+
+ [LINK](https://next.rl-institut.de/s/LJXYzBDyC5aDJtz) zu Rohdaten
+
+
+# Aufgaben
+☐ ☑ ☒
+   ☐ Metadaten zu den CSV-Dateien "HTW-Wetter" und "Sonja-Messdaten" generieren [ALL]
+   ☐ die richtigen Datenbankbefehle für die Fehler -> Lücken finden und Lücken schließen [ALL]
```

Comment:
add additional lines inbetween

Reply:
Write as following:

    - [ ]  Text
    - [x] Text 2

to create Checkboxes. It will look like this:

- [ ] Text
- [x] Text 2